backend/src/openarchiefbeheer/zaken/tasks.py

```diff
@@ -38,11 +37,15 @@ def retrieve_and_cache_zaken_from_openzaak() -> None:
         "einddatum__isnull": False,
         "einddatum__lt": today.isoformat(),
     }
-    if zaken_in_db := Zaak.objects.exists():
+
+    if not is_full_resync and Zaak.objects.exists():
         result = Zaak.objects.aggregate(Max("einddatum"))
         query_params.update({"einddatum__gt": result["einddatum__max"].isoformat()})
 
-    with zrc_client:
+    with transaction.atomic(), zrc_client, selectielijst_api_client or NoClient():
+        if is_full_resync:
+            Zaak.objects.all().delete()
```

Comment:
Do we need to delete destruction list items as well?

Reply:
No! The items have a _zaak_url field which will be used to re-link them to the new zaak if it exists. They only need to be deleted if no the zaak to which they were linked to is no longer present. This check happens in `resync_items_and_zaken()`